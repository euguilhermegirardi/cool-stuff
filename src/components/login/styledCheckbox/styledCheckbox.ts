import { CheckboxProps } from '@mui/material/Checkbox';
import { styled } from '@mui/material/styles';
import CheckedItem from 'assets/icons/checkedItem.svg';

export const MuiCheckBox = styled('span')(({ theme }) => ({
  width: 15,
  height: 15,
  border: `1px solid ${theme.palette.primary.main}`,
  backgroundColor: theme.palette.common.white,
}));

export const MuiCheckedItemBox = styled(MuiCheckBox)<CheckboxProps>(
  ({ theme }) => ({
    backgroundImage: `${theme.palette.primary.main}`,
    '&:before': {
      display: 'block',
      width: 15,
      height: 15,
      backgroundImage: `url(${CheckedItem})`,
      backgroundRepeat: 'no-repeat',
      backgroundPosition: 'center',
      backgroundSize: '13px 13px',
      content: '""',
    },
  })
);
