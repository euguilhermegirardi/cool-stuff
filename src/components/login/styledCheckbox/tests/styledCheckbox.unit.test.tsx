import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { MuiCheckedItemBox } from '../styledCheckbox';

test('it works', () => {
  const tree = renderer.create(<MuiCheckedItemBox />).toJSON();
  expect(tree).toMatchSnapshot();
})
