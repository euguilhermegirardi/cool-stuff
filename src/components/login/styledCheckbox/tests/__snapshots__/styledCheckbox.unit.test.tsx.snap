// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`it works 1`] = `
<span
  className="css-1gfkhh6"
/>
`;
